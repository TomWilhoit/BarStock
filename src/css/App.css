/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE: 
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  /* background: #1d2023; */
}

body {
  height: 100%;
}

/* #endRegion Global */



.Header-container {
  padding-bottom: 2em;
}

.Footer-container {
  position: relative;
  width: 100%;
  text-align: center;
  /* background: #1d2023; */
}

