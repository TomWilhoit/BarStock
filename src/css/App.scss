/********************
FONTS
font-family: 'Roboto Condensed', sans-serif;
font-family: 'Open Sans Condensed', sans-serif;
font-family: 'Abel', sans-serif;
font-family: 'Open Sans', sans-serif;

COLORS
Orange #fdba35
Blue Teal #3ebab4
Medium Gray #5e666c

NEW PALETTE:
Dark Gold: #EEAD0E
Light Gold: #F8DE9E
Dark Gray: #bbbbbb
Light Gray: #cfcfcf
Brown: #7d6c54
Black: #1d2023
White: #ffffff

*********************/

/* #region Global */
html {
  height: 100%;
  background: #dedede;
  background: url(./dust_scratches.png);
}

body {
  height: 100%;
}

/* #endRegion Global */

#root {
  height: 100%;
}

.App {
  height: 100%;
}

.All-Content {
  height: 100%;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}


.Header-container {
  margin-bottom: 2em;
}


/* #region Content Container  */


.content-container {
  display: table;
  max-width: 1200px;
  width: 100%;
  margin: auto;
}

.Inventory-container {
  display: table-cell;
  width: 60%;
  padding: 0em 0em 0em 2em;
}

.Totals-container {
  display: table-cell;
  width: 40%;
  padding: 0em 2em 0em 2em;
  vertical-align: bottom;
}

/* #endRegion Content Container */


.Submit-order {
  button {
    width: 100%;
    border-radius: 4px;
    outline: none;
    background: #EEAD0E;
    color: #fff;
    text-transform: uppercase;
    font-weight: 100;
    font-size: 1em;
    padding: .2em;
    font-family: 'Abel', sans-serif;
    cursor: pointer;
    &.disabled {
      pointer-events: none;
      opacity: .2;
    }
    &.enabled {
      pointer-events: all;
      opacity: 1;
    }
  }
}


.Footer-container {
  width: 100%;
  text-align: center;
  margin-top: 2em;
}



@media only screen and (max-width: 800px) {
  .content-container {
    display: block;
  }
  .Inventory-container {
    margin: auto;
    display: block;
    width: 100%;
    max-width: 600px;
    padding: 0;
    padding-bottom: 2em;
  }
  .Totals-container {
    display: block;
    margin: auto;
    width: 100%;
    padding: 0;
    max-width: 600px;
  }
}
