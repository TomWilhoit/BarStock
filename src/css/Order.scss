$font-title: 'Abel', sans-serif;
$font-copy: 'Lato', sans-serif;
$dark-blue: #11212a;
$btn-orange: #EEAD0E;
$dark-gray: #777777;
$white: #fff;

.Order-container {
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  justify-content: center;
  height: 100%;
  width: 100%;
  position: absolute;
  background-image: url(./order_bg.jpg);
  background-repeat: no-repeat;
  background-size: cover;
  background-position: top center;
  z-index: 9999999999;

  &::after {
    content: "";
    position: absolute;
    width: 100%;
    height: 100%;
    background: rgba(35, 31, 32, .6);
  }

  .Order {
    margin: auto;
    display: flex;
    flex-direction: column;
    z-index: 999;

    h3 {
      margin: 0;
      background-color: $dark-blue;
      color: $white;
      text-transform: uppercase;
      font-family: $font-title;
      display: flex;
      align-items: center;
      height: 50px;
      padding-left: 1em;
    }

    .order-form {
      width: 300px;
      background: $white;
      max-height: 350px;
      overflow-y: scroll;

      li {
        list-style-type: none;
        padding: .5em 1em;
        border-bottom: #cccccc solid 1px;

        h5 {
          margin: auto;
          font-family: $font-copy;
          font-size: .8em;
          font-weight: 100;
          color: $dark-gray;
        }
      }
    }

    .thank-you {
      color: $white;
    }

    .order-button,
    .back-button {
      margin-top: 1em;
      height: 30px;
      border-radius: 5px;
      font-size: 1rem;
      background-color:	$btn-orange;
      color: $white;
      outline: none;
      border: none;
      font-family: $font-title;
    }
  }
}
