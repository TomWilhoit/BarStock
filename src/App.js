import React, { Component } from "react";
import "./css/App.scss";
import Inventory from "./Inventory.js";
import Totals from "./Totals.js";
import Header from "./Header.js";
import Footer from "./Footer.js";
import Login from "./Login.js";
import Order from "./Order.js";

class App extends Component {
  constructor() {
    super();
    this.state = {
      loginDisplay: true,
      allDistributors: [],
      allInventory: [],
      allMenu: [],
      cartItems: [],
      totalCost: 0,
      currentUser: "",
      finalOrder: false
    };
  }

  componentDidMount() {
    fetch("http://whateverly-datasets.herokuapp.com/api/v1/distributor")
      .then(response => response.json())
      .then(supplier => {
        this.setState({
          allInventory: supplier.distributor[0].inventory,
          allDistributors: supplier.distributor[0]
        });
      })
      .catch(error => {
        throw new Error(error);
      });

    fetch("http://whateverly-datasets.herokuapp.com/api/v1/menu")
      .then(response => response.json())
      .then(products => {
        this.setState({
          allMenu: products.menu
        });
      })
      .catch(error => {
        throw new Error(error);
      });
  }

  changeCart = (item, math) => {
    const allCartItems = this.state.cartItems;
    let totalCost = this.state.totalCost;
    if (math === "plusOne") {
      totalCost += item.price;
      allCartItems.push(item);
      this.setState({
        cartItems: allCartItems,
        totalCost: totalCost
      });
    } else {
      const foundItem = allCartItems.findIndex(cartItem => {
        return cartItem === item;
      });
      totalCost -= allCartItems[foundItem].price;
      allCartItems.splice(foundItem, 1);
      this.setState({
        cartItems: allCartItems,
        totalCost: totalCost
      });
    }
  };

  toggleLogin = () => {
    if (this.state.loginDisplay === true) {
      this.setState({ loginDisplay: false });
    } else if (this.state.loginDisplay === false) {
      this.setState({ loginDisplay: true });
    }
  };

  submitOrder = () => {
    this.setState({
      finalOrder: true
    });
  };

  backToCart = () => {
    this.setState({
      finalOrder: false,
      loginDisplay: false
    });
  };

  loginAccount = userNameValue => {
    this.setState({
      currentUser: userNameValue
    });
  };

  render() {
    let submitClass = "disabled";
    this.state.cartItems.length >= 1
      ? (submitClass = "enabled")
      : (submitClass = "disabled");

    if (this.state.loginDisplay === true) {
      return (
        <div className="App">
          <Login
            loginDisplay={this.state.loginDisplay}
            toggleLogin={this.toggleLogin}
            loginAccount={this.loginAccount}
          />
        </div>
      );
    } else if (this.state.finalOrder === true) {
      return (
        <div className="App">
          <Order
            cartItems={this.state.cartItems}
            user={this.state.currentUser}
            finalTotal={this.state.totalCost}
            backToCart={this.backToCart}
          />
        </div>
      );
    } else {
      return (
        <div className="App">
          <div className="All-Content">
            <div className="Header-container">
              <Header currentUser={this.state.currentUser} />
            </div>
            <div className="content-container">
              <div className="Inventory-container">
                <Inventory
                  allInventory={this.state.allInventory}
                  changeCart={this.changeCart}
                  cartItems={this.state.cartItems}
                />
              </div>
              <div className="Totals-container">
                <Totals
                  cartMenu={this.state.allMenu}
                  cartItems={this.state.cartItems}
                  changeCart={this.changeCart}
                />
                <section className="Submit-order">
                  <button className={submitClass} onClick={this.submitOrder}>
                    Checkout
                  </button>
                </section>
              </div>
            </div>
            <div className="Footer-container">
              <Footer />
            </div>
          </div>
        </div>
      );
    }
  }
}

export default App;
