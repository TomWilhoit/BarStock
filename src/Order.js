import React, { Component } from 'react';
import App from './App.js';
import BarLogo from './BarStock_Logo.png'
import './css/Order.scss';

class Order extends Component {
  constructor(props) {
    super(props);
    this.state = {
      orderComplete: false
    }
  }

  displayObjects = () => {
    let sumCart = this.props.cartItems.reduce((acc, el) => {
      if(!acc[el.product]){
        acc[el.product] = 0
      }
      acc[el.product]++
      return acc
    }, {});
    return Object.keys(sumCart).map((prod, index) => {
      let thisProd = this.props.cartItems.find(ac => ac.product === prod)
      return <li className="orders-list" key={index}>
        <h5 key={index}>{sumCart[prod]} x {prod} = Total: ${(thisProd.price * sumCart[prod]).toFixed(2)}</h5>
      </li>
    })
  }

  send = () => {
  //   var link = "mailto:michaelryankrog@gmail.com"
  //          + "&subject=" + escape("This is my subject")
  //          + "&body=" + escape(document.getElementById('myText').innerText)
  // ;
  //   window.location.href = link;
    this.setState({
      orderComplete: true
    })
  }


  render() {
    if (this.state.orderComplete === true) {
      return (
        <div className="Order">
          <h3>Thanks for your order!</h3>
        </div>
      )} else {
      return (
        <div className="Order-container">
          <form className="Order">
            <h3 className="order-quote">Confirmation {this.props.user}</h3>
            <div id="myText" className="order-form">
              {this.displayObjects()}
            </div>
            <h3>Total ${this.props.finalTotal.toFixed(2)}</h3>
            <button className="order-button" onClick={this.send}>Submit Order</button>
            <button className="back-button" onClick={this.props.backToCart}>Back to Cart</button>
          </form>
        </div>
      );
    }
  }
};



export default Order;
