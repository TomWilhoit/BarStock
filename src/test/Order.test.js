import React from "react";
import App from "../App";
import { shallow } from "enzyme";
import { Order } from "../Order";

describe("Order", () => {
  let wrapper;
  let cartItems = [];
  let user = "";
  const mockFunc = jest.fn();
  let finalTotal = 0;

  beforeEach(() => {
    wrapper = shallow(
      <Order
        finalTotal={finalTotal}
        cartItems={cartItems}
        user={user}
        displayObjects={mockFunc}
        submitOrder={mockFunc}
      />
    );
  });

  it("should match snapshot when all data is passed correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should change the state of orderComplete when send is invoked", () => {
    expect(wrapper.state("orderComplete")).toEqual(false);
    wrapper.instance().send();
    expect(wrapper.state("orderComplete")).toEqual(true);
  });
});
