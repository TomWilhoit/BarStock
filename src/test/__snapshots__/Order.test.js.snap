// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order should match snapshot when all data is passed correctly 1`] = `
<div
  className="Order-container"
>
  <form
    className="Order"
  >
    <h3
      className="order-quote"
    >
      Confirmation 
    </h3>
    <div
      className="order-form"
      id="myText"
    />
    <h3>
      Total $
      0.00
    </h3>
    <button
      className="order-button"
      onClick={[Function]}
    >
      Submit Order
    </button>
    <button
      className="back-button"
    >
      Back to Cart
    </button>
  </form>
</div>
`;
